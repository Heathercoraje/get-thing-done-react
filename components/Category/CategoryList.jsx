import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';


const Wrapper = styled.div`
	padding-left: 0.6rem;
	color: white;
`;
const CategoryList = ({ categories, ...props }) => (
	<Wrapper className="category-list">
		{categories.map(category => (
    <CategoryItem key={category.id} {...props} {...category} />
		))}
  </Wrapper>
);

CategoryList.propTypes = {
	props: PropTypes.object,
	categories: PropTypes.array,
	category: PropTypes.object,
	id: PropTypes.number
}
const CategoryItem = props => (
	<div className="category-div">
		<li
			onClick={props.click}
			value={props.id}
			data-name={props.name}
			className="category-text"
		>
			{props.name}
  </li>
		<button className="button-delete" onClick={props.delete} value={props.id}>
			&#8854;
  </button>
  </div>
);
CategoryItem.propTypes = {
	id: PropTypes.number,
	name: PropTypes.string,
	click: PropTypes.func,
	delete: PropTypes.func
}

export { CategoryList, CategoryItem };
