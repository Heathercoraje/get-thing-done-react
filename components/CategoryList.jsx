import React from 'react';
// they are just all function with return value
// which happens to be a component, simple like that

const CategoryList = ({ categories, deleteCategory, clickCate, selectedCategory }) => {
  const targetCategories = categories.slice(1);
  const Renderlist = categories =>
    targetCategories.map(category => (
      <div key={category.id}>
        <li onClick={clickCate} id={category.id}>
          {category.category}
        </li>
        <button onClick={deleteCategory} value={category.id}>
					Delete
        </button>
      </div>
    ));
  return (
    <div className="left">
      <li onClick={clickCate} id="0">
				Todo
      </li>
      <div>{Renderlist(categories)}</div>
    </div>
  );
};

export default CategoryList;
