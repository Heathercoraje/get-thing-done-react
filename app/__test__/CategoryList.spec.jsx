import React from 'react';
import { shallow } from 'enzyme';
import CategoryList from '../../components/CategoryList';

test('renders correctly', () => {
  const categories = [{ id: 0, name: 'Todo', desc: 'Get things done today' }];
  const component = shallow(<CategoryList categories={categories} />);
  expect(component).toMatchSnapshot();
});
