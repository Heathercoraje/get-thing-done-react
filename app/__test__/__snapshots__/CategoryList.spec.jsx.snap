// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <div
        className="category-div"
    >
        <li
            className="category-text"
            onClick={undefined}
            value="0"
        >
            Todo
        </li>
    </div>
    <div>
        
    </div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <div
            className="category-div"
      >
            <li
                  className="category-text"
                  onClick={undefined}
                  value="0"
            >
                  Todo
            </li>
      </div>
      <div>
            
      </div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CategoryList
        categories={
                Array [
                        Object {
                          "desc": "Get things done today",
                          "id": 0,
                          "name": "Todo",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "categories": Array [
            Object {
              "desc": "Get things done today",
              "id": 0,
              "name": "Todo",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <div
                    className="category-div"
          >
                    <li
                              className="category-text"
                              onClick={undefined}
                              value="0"
                    >
                              Todo
                    </li>
          </div>
          <div>
                    
          </div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <div
                    className="category-div"
          >
                    <li
                              className="category-text"
                              onClick={undefined}
                              value="0"
                    >
                              Todo
                    </li>
          </div>
          <div>
                    
          </div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CategoryList
    categories={
        Array [
            Object {
              "desc": "Get things done today",
              "id": 0,
              "name": "Todo",
            },
          ]
    }
/>,
}
`;
